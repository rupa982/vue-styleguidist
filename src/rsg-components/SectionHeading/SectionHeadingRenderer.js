import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';

export function SectionHeadingRenderer({
	classes,
	children,
	toolbar,
	id,
	href,
	depth,
	deprecated,
}) {
	const headingLevel = Math.min(6, depth);
	const Tag = `h${headingLevel}`;
	const headingClasses = cx(classes.heading, classes[`heading${headingLevel}`], {
		[classes.isDeprecated]: deprecated,
	});
	return (
		<Tag id={id} className={classes.root}>
			<a href={href} className={headingClasses}>
				{children}
			</a>
			<div className={classes.toolbar}>{toolbar}</div>
		</Tag>
	);
}

export const styles = ({ color, space, fontSize, fontFamily }) => ({
	root: {
		display: 'flex',
		marginBottom: space[1],
		alignItems: 'center',
	},
	heading: {
		color: color.base,
		fontFamily: fontFamily.base,
		fontWeight: 'normal',
		'&:hover, &:active': {
			isolate: false,
			textDecoration: 'underline',
		},
	},
	heading1: {
		fontSize: fontSize.h1,
	},
	heading2: {
		fontSize: fontSize.h2,
	},
	heading3: {
		fontSize: fontSize.h3,
	},
	heading4: {
		fontSize: fontSize.h4,
	},
	heading5: {
		fontSize: fontSize.h5,
	},
	heading6: {
		fontSize: fontSize.h6,
	},
	isDeprecated: {
		textDecoration: 'line-through',
		color: color.light,
	},
	toolbar: {
		marginLeft: 'auto',
	},
});

SectionHeadingRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node,
	toolbar: PropTypes.node,
	id: PropTypes.string.isRequired,
	href: PropTypes.string.isRequired,
	depth: PropTypes.number.isRequired,
	deprecated: PropTypes.bool,
};

export default Styled(styles)(SectionHeadingRenderer);
