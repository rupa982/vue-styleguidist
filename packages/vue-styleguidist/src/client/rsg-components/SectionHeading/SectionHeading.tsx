import React from 'react'
import PropTypes from 'prop-types'
import Slot from 'rsg-components/Slot'
import SectionHeadingRenderer from 'rsg-components/SectionHeading/SectionHeadingRenderer'

const localPropTypes = {
	children: PropTypes.node,
	id: PropTypes.string.isRequired,
	slotName: PropTypes.string.isRequired,
	slotProps: PropTypes.shape({
		isolated: PropTypes.bool.isRequired,
		parentComponent: PropTypes.shape({
			href: PropTypes.string.isRequired,
			visibleName: PropTypes.string.isRequired
		})
	}).isRequired,
	href: PropTypes.string,
	depth: PropTypes.number.isRequired,
	deprecated: PropTypes.bool,
	pagePerSection: PropTypes.bool
}

type SectionHeadingProps = PropTypes.InferProps<typeof localPropTypes>

const SectionHeading: React.FunctionComponent<SectionHeadingProps> = ({
	slotName,
	slotProps,
	children,
	id,
	...rest
}) => {
	const parentHref = slotProps.parentComponent && slotProps.parentComponent.href

	return (
		<SectionHeadingRenderer
			toolbar={slotProps.parentComponent ? null : <Slot name={slotName} props={slotProps} />}
			id={id}
			{...rest}
			parentName={slotProps.parentComponent && slotProps.parentComponent.visibleName}
			parentHref={parentHref}
		>
			{children}
		</SectionHeadingRenderer>
	)
}

SectionHeading.propTypes = localPropTypes

export default SectionHeading
