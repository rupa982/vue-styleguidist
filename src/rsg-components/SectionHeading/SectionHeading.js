import React from 'react';
import PropTypes from 'prop-types';
import Slot from 'rsg-components/Slot';
import SectionHeadingRenderer from 'rsg-components/SectionHeading/SectionHeadingRenderer';
import { getUrlNavigation } from '../../utils/utils';

export default function SectionHeading(
	{ slotName, slotProps, children, id, level, name, nameParent, collection = {}, ...rest },
	{ config }
) {
	const navigation = config.navigation;
	const href = getUrlNavigation(navigation, {
		level,
		sections: collection.sections,
		components: collection.components,
		nameParent,
		name,
		slug: id,
	});
	return (
		<SectionHeadingRenderer
			toolbar={<Slot name={slotName} props={slotProps} />}
			id={id}
			navigation={navigation}
			href={href}
			{...rest}
		>
			{children}
		</SectionHeadingRenderer>
	);
}

SectionHeading.propTypes = {
	children: PropTypes.node,
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	nameParent: PropTypes.string.isRequired,
	level: PropTypes.number.isRequired,
	slotName: PropTypes.string.isRequired,
	slotProps: PropTypes.object.isRequired,
	depth: PropTypes.number.isRequired,
	deprecated: PropTypes.bool,
	collection: PropTypes.object,
};

SectionHeading.contextTypes = {
	config: PropTypes.object,
};
