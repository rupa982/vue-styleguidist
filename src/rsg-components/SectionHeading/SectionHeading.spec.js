import React from 'react';
import SectionHeading, { SectionHeadingRenderer, styles } from './SectionHeadingRenderer';

const props = {
	classes: classes(styles),
	id: 'heading',
	href: '/heading',
	depth: 3,
};

it('should pass props to slots', () => {
	const actual = shallow(
		<SectionHeading
			id="heading"
			depth={1}
			href="/heading"
			slotName="slot"
			slotProps={{ foo: 1, bar: 'baz' }}
		>
			Heading
		</SectionHeading>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render H3 tag for depth of 3', () => {
	const actual = shallow(<SectionHeadingRenderer {...props}>Heading</SectionHeadingRenderer>);

	expect(actual).toMatchSnapshot();
});

it('renderer should render H1 tag if depth of 1', () => {
	const actual = shallow(
		<SectionHeadingRenderer {...props} depth={1}>
			Heading
		</SectionHeadingRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render H6 tag if depth is over 6', () => {
	const actual = shallow(
		<SectionHeadingRenderer {...props} depth={7}>
			Heading
		</SectionHeadingRenderer>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render heading with deprecated styles', () => {
	const actual = shallow(
		<SectionHeadingRenderer {...props} deprecated>
			Heading
		</SectionHeadingRenderer>
	);

	expect(actual).toMatchSnapshot();
});
