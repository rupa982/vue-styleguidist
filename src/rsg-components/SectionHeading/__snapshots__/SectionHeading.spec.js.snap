// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render H1 tag if depth of 1 1`] = `
<h1
  className="root"
  id="heading"
>
  <a
    className="heading heading1"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h1>
`;

exports[`renderer should render H3 tag for depth of 3 1`] = `
<h3
  className="root"
  id="heading"
>
  <a
    className="heading heading3"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h3>
`;

exports[`renderer should render H6 tag if depth is over 6 1`] = `
<h6
  className="root"
  id="heading"
>
  <a
    className="heading heading6"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h6>
`;

exports[`renderer should render heading with deprecated styles 1`] = `
<h3
  className="root"
  id="heading"
>
  <a
    className="heading heading3 isDeprecated"
    href="/heading"
  >
    Heading
  </a>
  <div
    className="toolbar"
  />
</h3>
`;

exports[`should pass props to slots 1`] = `
<SectionHeadingRenderer
  classes={
    Object {
      ".rsg--heading-1:hover, .rsg--heading-1:active": "rsg--heading-1:hover, .rsg--heading-1:active",
      "heading": "rsg--heading-1",
      "heading1": "rsg--heading1-2",
      "heading2": "rsg--heading2-3",
      "heading3": "rsg--heading3-4",
      "heading4": "rsg--heading4-5",
      "heading5": "rsg--heading5-6",
      "heading6": "rsg--heading6-7",
      "isDeprecated": "rsg--isDeprecated-8",
      "root": "rsg--root-0",
      "toolbar": "rsg--toolbar-9",
    }
  }
  depth={1}
  href="/heading"
  id="heading"
  slotName="slot"
  slotProps={
    Object {
      "bar": "baz",
      "foo": 1,
    }
  }
>
  Heading
</SectionHeadingRenderer>
`;
